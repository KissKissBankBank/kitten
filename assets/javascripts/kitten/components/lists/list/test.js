import React from 'react'
import renderer from 'react-test-renderer'
import { List } from './index'
import { shallow } from 'enzyme'

describe('<List>', () => {
  describe('by default', () => {
    it('matches with snapshot', () => {
      const component = renderer
        .create(
          <List>
            <div>Item 1</div>
            <div>Item 2</div>
            <div>Item 3</div>
          </List>,
        )
        .toJSON()

      expect(component).toMatchSnapshot()
    })
  })

  describe('with <ListButtonItem>', () => {
    describe('by default', () => {
      it('matches with snapshot', () => {
        const component = renderer
          .create(
            <List>
              <List.ButtonItem withTopBorder>Alice</List.ButtonItem>
              <List.ButtonItem>Cheshire Cat</List.ButtonItem>
              <List.ButtonItem>Mad Hatter</List.ButtonItem>
            </List>,
          )
          .toJSON()

        expect(component).toMatchSnapshot()
      })

      it('trigger the onClick prop', () => {
        const handleClick = jest.fn()
        const component = shallow(
          <List>
            <List.ButtonItem withTopBorder>Alice</List.ButtonItem>
            <List.ButtonItem onClick={handleClick}>
              Cheshire Cat
            </List.ButtonItem>
            <List.ButtonItem>Mad Hatter</List.ButtonItem>
          </List>,
        )

        const button = component.find(List.ButtonItem).at(1).dive()

        button.simulate('click')

        expect(handleClick).toHaveBeenCalled()
        handleClick.mockRestore()
      })
    })

    describe('with a disabled button', () => {
      it('matches with snapshot', () => {
        const component = renderer
          .create(
            <List>
              <List.ButtonItem withTopBorder>Alice</List.ButtonItem>
              <List.ButtonItem disabled>Cheshire Cat</List.ButtonItem>
              <List.ButtonItem>Mad Hatter</List.ButtonItem>
            </List>,
          )
          .toJSON()

        expect(component).toMatchSnapshot()
      })

      it('does not trigger the onClick prop', () => {
        const handleClick = jest.fn()
        const component = shallow(
          <List>
            <List.ButtonItem withTopBorder>Alice</List.ButtonItem>
            <List.ButtonItem disabled onClick={handleClick}>
              Cheshire Cat
            </List.ButtonItem>
            <List.ButtonItem>Mad Hatter</List.ButtonItem>
          </List>,
        )

        const button = component.find(List.ButtonItem).at(1).dive()

        button.simulate('click')

        expect(handleClick).not.toHaveBeenCalled()
      })
    })
  })
})
