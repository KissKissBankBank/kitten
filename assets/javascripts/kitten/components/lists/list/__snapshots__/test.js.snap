// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List> by default matches with snapshot 1`] = `
<ul
  className="list__StyledList-sc-1ns0hml-0 hEwMDn"
>
  <li>
    <div>
      Item 1
    </div>
  </li>
  <li>
    <div>
      Item 2
    </div>
  </li>
  <li>
    <div>
      Item 3
    </div>
  </li>
</ul>
`;

exports[`<List> with <ListButtonItem> by default matches with snapshot 1`] = `
<ul
  className="list__StyledList-sc-1ns0hml-0 hEwMDn"
>
  <li>
    <div
      className="button-item__Item-sc-1xwobvm-0 eQyeg"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="button-item__Content-sc-1xwobvm-1 fKXndd"
      >
        Alice
      </div>
      <div
        className="button-item__Arrow-sc-1xwobvm-2 jcFtTs"
      >
        <svg
          className="k-ArrowIcon k-ArrowIcon--right k-Button__icon button-item__ArrowIcon-sc-1xwobvm-3 iooXxv"
          data-radium={true}
          viewBox="0 0 6 6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Arrow
          </title>
          <path
            d="M6 0H0v6h2V2h4z"
          />
        </svg>
      </div>
    </div>
  </li>
  <li>
    <div
      className="button-item__Item-sc-1xwobvm-0 cXshSK"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="button-item__Content-sc-1xwobvm-1 fKXndd"
      >
        Cheshire Cat
      </div>
      <div
        className="button-item__Arrow-sc-1xwobvm-2 jcFtTs"
      >
        <svg
          className="k-ArrowIcon k-ArrowIcon--right k-Button__icon button-item__ArrowIcon-sc-1xwobvm-3 iooXxv"
          data-radium={true}
          viewBox="0 0 6 6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Arrow
          </title>
          <path
            d="M6 0H0v6h2V2h4z"
          />
        </svg>
      </div>
    </div>
  </li>
  <li>
    <div
      className="button-item__Item-sc-1xwobvm-0 cXshSK"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="button-item__Content-sc-1xwobvm-1 fKXndd"
      >
        Mad Hatter
      </div>
      <div
        className="button-item__Arrow-sc-1xwobvm-2 jcFtTs"
      >
        <svg
          className="k-ArrowIcon k-ArrowIcon--right k-Button__icon button-item__ArrowIcon-sc-1xwobvm-3 iooXxv"
          data-radium={true}
          viewBox="0 0 6 6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Arrow
          </title>
          <path
            d="M6 0H0v6h2V2h4z"
          />
        </svg>
      </div>
    </div>
  </li>
</ul>
`;

exports[`<List> with <ListButtonItem> with a disabled button matches with snapshot 1`] = `
<ul
  className="list__StyledList-sc-1ns0hml-0 hEwMDn"
>
  <li>
    <div
      className="button-item__Item-sc-1xwobvm-0 eQyeg"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="button-item__Content-sc-1xwobvm-1 fKXndd"
      >
        Alice
      </div>
      <div
        className="button-item__Arrow-sc-1xwobvm-2 jcFtTs"
      >
        <svg
          className="k-ArrowIcon k-ArrowIcon--right k-Button__icon button-item__ArrowIcon-sc-1xwobvm-3 iooXxv"
          data-radium={true}
          viewBox="0 0 6 6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Arrow
          </title>
          <path
            d="M6 0H0v6h2V2h4z"
          />
        </svg>
      </div>
    </div>
  </li>
  <li>
    <div
      className="button-item__Item-sc-1xwobvm-0 hDlIEc"
      disabled={true}
      onBlur={[Function]}
      onClick={null}
      onFocus={[Function]}
      onKeyPress={null}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="button-item__Content-sc-1xwobvm-1 fKXndd"
      >
        Cheshire Cat
      </div>
      <div
        className="button-item__Arrow-sc-1xwobvm-2 jcFtTs"
      >
        <svg
          className="k-ArrowIcon k-ArrowIcon--right k-ArrowIcon--disabled k-Button__icon button-item__ArrowIcon-sc-1xwobvm-3 IIVFy"
          data-radium={true}
          viewBox="0 0 6 6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Arrow
          </title>
          <path
            d="M6 0H0v6h2V2h4z"
          />
        </svg>
      </div>
    </div>
  </li>
  <li>
    <div
      className="button-item__Item-sc-1xwobvm-0 cXshSK"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="button-item__Content-sc-1xwobvm-1 fKXndd"
      >
        Mad Hatter
      </div>
      <div
        className="button-item__Arrow-sc-1xwobvm-2 jcFtTs"
      >
        <svg
          className="k-ArrowIcon k-ArrowIcon--right k-Button__icon button-item__ArrowIcon-sc-1xwobvm-3 iooXxv"
          data-radium={true}
          viewBox="0 0 6 6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Arrow
          </title>
          <path
            d="M6 0H0v6h2V2h4z"
          />
        </svg>
      </div>
    </div>
  </li>
</ul>
`;
